mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,192 @@
+
+    def plot(self):
+        """
+        Computes and returns a list of per sample Het/Hom ratios suitable for plotting
+
+        Returns:
+          a list containing the Het/Hom ratio statistic for each sample in the dataset
+        """
+        hetHomRatio = [ float(self.hets[sampleid])/float(self.homs[sampleid]) for sampleid in self.homs.keys() ]
+        return hetHomRatio
+
+
+class GenotypeCallRatesDistribution(object):
+    """ GenotypeCallRatesDistribution class.
+    GenotypeCallRatesDistribution computes a distribution of per-sample genotype call rates from a genotypeRDD.
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes a GenotypeCallRatesDistribution class.
+        Retrieves counts of called and missing genotypes from a genotypeRDD.
+
+        Args:
+            :param ss: SparkContext
+            :param genotypeRDD: bdgenomics genotypeRDD
+        """
+
+        df =  genotypeRDD.toDF()
+        called_df =  df.where( (df.alleles[0] != u'NO_CALL')).groupBy("sampleid").count().collect()
+        self.called_map =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in called_df }
+
+        missing_df =  df.where( (df.alleles[0] == u'NO_CALL')).groupBy("sampleid").count().collect()
+        self.missing_map =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in missing_df }
+
+    def plot(self):
+        """
+        Computes and returns a list of per sample call rates suitable for plotting.
+
+        Returns:
+           A list of per sample call rates.
+        """
+        callRatePerSample = [ float(self.called_map[sampleid])/ \
+                              ( float(self.missing_map[sampleid])+float(self.called_map[sampleid])) \
+                              for sampleid in self.called_map.keys() ]
+
+        return callRatePerSample
+
+
+class QualityByDepthDistribution(object):
+    """ QualityByDepthDistribution class.
+    QualityByDepthDistribution calculates the Quality By Depth (QD) score distribution from a variantRDD.
+    """
+    def __init__(self,ss,variantRDD):
+        """
+        Initializes a QualityByDepthDistribution class.
+        Computes the distribution of QD scores from a variantRDD.
+
+        Args:
+            :param ss: SparkContext
+            :param variantRDD: bdgenomics variantRDD
+        """
+
+        # bin QD score by 0.1 QD units by take the floor(x*10) / 10
+        binConstant = 10
+
+        self.data = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).map(
+            lambda x: (math.floor(float(x['attributes']['QD'])*binConstant)/binConstant)).countByValue()
+
+    def plot(self):
+        """
+        Computes and returns binned QD score plot data suitable for plotting as a Counter object.
+
+        Returns:
+            Computed binned QD score data as a Counter object.
+        """
+        return Counter(dict(self.data.items()))
```

Comment:
Add unit tests, also a nbconvert in make test that converts the notebook and tests them